compiler/rustc_builtin_macros/src/standard_library_imports.rs

```diff
@@ -44,20 +44,37 @@ pub fn inject(
 
     // .rev() to preserve ordering above in combination with insert(0, ...)
     for &name in names.iter().rev() {
-        let ident = if edition >= Edition2018 {
-            Ident::new(name, span)
-        } else {
-            Ident::new(name, call_site)
+        let make_ident = |name| {
+            if edition >= Edition2018 {
+                Ident::new(name, span)
+            } else {
+                Ident::new(name, call_site)
+            }
         };
-        krate.items.insert(
-            0,
-            cx.item(
-                span,
-                ident,
-                thin_vec![cx.attr_word(sym::macro_use, span)],
-                ast::ItemKind::ExternCrate(None),
-            ),
-        );
+        if name == sym::compiler_builtins {
+            // compiler_builtins is a private implementation detail. We only
+            // need to insert it into the crate graph for linking and should not
+            // expose any of its public API.
+            krate.items.insert(
+                0,
+                cx.item(
+                    span,
+                    make_ident(kw::Underscore),
```

Comment:
This will generate `extern crate _ as compiler_builtins;`, no?
How does it work?